test/common/index.js

```diff
@@ -528,7 +528,14 @@ let catchWarning;
 function expectWarning(nameOrMap, expected, code) {
   if (catchWarning === undefined) {
     catchWarning = {};
-    process.on('warning', (warning) => catchWarning[warning.name](warning));
+    process.on('warning', (warning) => {
+      if (!catchWarning[warning.name]) {
+        throw new TypeError(
+          `An "${warning.name}" was triggered without being expected.\n${inspect(warning)}`
```

Comment:
Micro-nit: `A` vs. `An` is tricky here. Maybe leave it off?

```suggestion
          `"${warning.name}" was triggered without being expected.\n${inspect(warning)}`
```

Reply:
Done